content/posts/cicd_ha.md

```diff
@@ -0,0 +1,132 @@
+
+In a *Jenkins enterprise experience* this is a must, and that is why for example, [CloudBees Core](https://docs.cloudbees.com/docs/cloudbees-core/latest/) (usually known as *Jenkins Enterprise*) provides out of the box these capabilities with master management features and Kubernetes scalability.
+
+### The Cloud Native evolution (a.k.a. Jenkins X)
+
+But if we go purely Kubernetes native, and we provide an architecture where every piece of the pipeline orchestration and execution is just a state declaration object that can be recovered automatically when something goes down, then the HA and resilient experience is event better.
+
+ [Jenkins X](https://jenkins-x.io) takes these concepts and all the previous mentioned best practices of scalability and resilience. Everything is about Kubernetes native objects definitions (based on [CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)), which are resilient by nature from Kubernetes. No need to think about decoupled pipelines, infrastructure replication, or recovering scenarios. It is already out of the box from its solution architecture design. If you deploy your Kubernetes platform with some [Disaster Recovery](https://en.wikipedia.org/wiki/Disaster_recovery) configurations like Cloud cross-region availability, autoscaling pools and backup features (I think [Velero](https://velero.io/) is an interesting solution for K8s backup), it is practically impossible to completely lose the service for your CI/CD pipelines (except the case the World is ending...).
+
+As an example, in the image below we can see that a pipeline execution on a default Jenkins X deployment recovered just in 30 seconds to continue running after a [Pod](https://kubernetes.io/docs/concepts/workloads/pods/pod/) failure simulation. Nothing needed to be done in terms of deployment or pipeline definition. It is just its native behavior.
+
+![Jenkins X pod recovery](/img/cicd-ha/jenkins-x_podrestart.png)
+
+## The message and conclusion
+
+I've received in the past several redundant requests from organizations like the following ones:
+
+* *"We need to replicate and load balance our masters to have HA"*
+* *"My masters need to be up and running all the time as Active-Active to have a continuously running CI/CD service"*
+* *"How can I write to my shared storage at the same time with replicated masters?"* (another active-active HA conversation)
+
+And many more similar.
+
+Funny thing is that lots of cases were not prioritizing scalability, pipeline and master decoupling, or containerized orchestrated deployments. And then you realize that most of these requests come from IT requirements definitions in terms of HA general software architectures, no matter the nature of the service. 
```

Comment:
Should add a "The" at the start of the sentence

Reply:
Done with all. Thanks for the corrections!